### ISSUE TITLE
The finding of the optimal radius of influence makes assumption on the ordering of the dimensions in the longitude array

### ISSUE THREAD

**Issue by adybbroe** [pytroll/pyresample#554]
#### Code Sample, a minimal, complete, and verifiable piece of code

See also comments in a recent Satpy PR: https://github.com/pytroll/satpy/pull/2565

```python
# Your code here

```
#### Problem description

[this should also explain **why** the current behaviour is a problem and why the 
expected output is a better solution.]

**Comment by djhoese** [comment]
As mentioned in your pull request we can't assume xarray is available in Pyresample. That is standard in Satpy, but not pyresample. If I'm understanding your PR correctly, your swath DataArrays have xarray dimensions `("x", "y")`? Is that correct? In that case, I have two possible solutions that I see:

1. Update the geocentric resolution to take pixels in the middle row first two columns like it is now *and* the middle column first two rows and take the largest resolution size found.
2. Update your reader in Satpy to be `("y", "x")`. While technically it shouldn't matter, I'm a little scared that this is only one piece of the Satpy/Pyresample puzzle that you're finding an issue with, but that we make the assumption of `("y", "x")` in many places.

**Comment by adybbroe** [comment]
Ok, thanks for the comments and suggestions. And, yes, the dimensions are `("x", "y")` which is non-standard, I know (see below). 

Concerning option 2: Yes, but actually in this case it is the data format which is "wrong", or non-standard. So, for now I rather keep the reader as it is, and wait for the agency (ESA) to fix the format. They have acknowledged the issue at least.

But, as the reader returns an xarray DataArray I thought it would be appropriate to actually try use that information, rather than make assumption on the data layout.

Having said that, I also like your option 1. So, would you prefer:

  1. Keep the suggestion above when the data is an xarray DataArray, and then improve the code for Numpy cases as you propose in your option 1? Or
  2. Skip using the xarray capability and improve as in your option 1, which will will work in all cases?

**Comment by adybbroe** [comment]
Ok, thanks for the comments and suggestions. And, yes, the dimensions are `("x", "y")` which is non-standard, I know (see below). 

Concerning option 2: Yes, but actually in this case it is the data format which is "wrong", or non-standard. So, for now I rather keep the reader as it is, and wait for the agency (ESA) to fix the format. They have acknowledged the issue at least.

But, as the reader returns an xarray DataArray I thought it would be appropriate to actually try use that information, rather than make assumption on the data layout.

Having said that, I also like your option 1. So, would you prefer:

  1. Keep the suggestion above when the data is an xarray DataArray, and then improve the code for Numpy cases as you propose in your option 1? Or
  2. Skip using the xarray capability and improve as in your option 1, which will will work in all cases?

**Comment by djhoese** [comment]
> Concerning option 2: Yes, but actually in this case it is the data format which is "wrong", or non-standard. So, for now I rather keep the reader as it is, and wait for the agency (ESA) to fix the format. They have acknowledged the issue at least.

I don't agree with this, but I'm not doing the coding so I can accept it. Everyone is admitting it is "wrong" (or at least unexpected) and the solution should be a simple `.T` on the array. Anyway...

> Having said that, I also like your option 1. So, would you prefer:
>
>   1. Keep the suggestion above when the data is an xarray DataArray, and then improve the code for Numpy cases as you propose in your option 1? Or
>   2. Skip using the xarray capability and improve as in your option 1, which will will work in all cases?

I prefer 2. It (at least theoretically) should only improve the accuracy of this method which is at best a guess. I'd prefer to avoid Xarray workarounds as much as possible, especially when the alternative is "make this functionality work better", but I could also be missing some complexity in this task so I could be convinced otherwise.

**Comment by gerritholl** [comment]
I agree that option 2 is preferable, but I don't have a strong opinion on this.

**Comment by gerritholl** [comment]
Is this still relevant, or have reader and data format been fixed?
